Expensify/App
# Enable active client manager, multiple tabs, and previous PR cleanup

- Cleans up some of the final comments that were in https://github.com/Expensify/ReactNativeChat/pull/426
- Enables the active client manager and multi-tab functionality to remove some TODOs in the code Fixes https://github.com/Expensify/Expensify/issues/141684
- Adds a warning for some undefined behavior with `Ion.connect()` and provides instructions on proper use

# Tests for multi-tab functionality
- Open multiple tabs to the chat application with the same USER A signed in
- Inspect localStorage to see that `activeClients` has multiple GUIDs 
- In a different browser (or in an incognito window), sign in to the chat application as USER B (make sure both users have access to the same reports)
- As USER B, send a comment to any chat that USER A is not currently looking at
- Verify there is a browser notification and clicking it should take you to one of the chat apps for USER A (the first tab you opened)
1. Go to the second tab for USER A and check the JS console (you need to have verbose logging turned on) and verify you see: `[NOTIFICATION] Skipping notification because this client is not the leader`
1. Now close the first tab for USER A
1. Go back to USER B and add a new comment
1. Verify there is a browser notification and clicking it should take you to the tab for USER A that is still open
